src/internal/ppsdb/ppsdb.go

```diff
@@ -45,7 +48,20 @@ var PipelineJobsOutputIndex = &col.Index{
 	},
 }
 
-var pipelineJobsIndexes = []*col.Index{PipelineJobsPipelineIndex, PipelineJobsOutputIndex}
+func PipelineJobTerminalKey(pipeline *pps.Pipeline, isTerminal bool) string {
+	return fmt.Sprintf("%s_%v", pipeline.Name, isTerminal)
+}
+
+// PipelineJobsOutputIndex maps job outputs to the PipelineJob that create them.
+var PipelineJobsTerminalIndex = &col.Index{
```

Comment:
Add a secondary index for pipeline jobs which sorts them by `[pipeline, isTerminal]`, so that workers can easily subscribe for unfinished jobs.

Reply:
So this isn't actually the same as the `multi` option we previously had in collections (which was the ability to have a row represented in an index multiple times based on an array field), this is more of a 'compound' or 'composite' index, which we are already using for the primary keys of commits and jobs.

Reply:
Oh, sorry, I meant compound indices, I wasn't aware of `multi` at all. I forgot that a bunch of our primary keys could also work that way, which could probably save us an index on those